tests/native_api_tests/test_api_model.cpp

```diff
@@ -513,4 +537,133 @@ TEST_CASE("API Model") {
+    std::vector<double> const node_u_rated{100.0};
+    Buffer node_buffer{PGM_def_input_node, 1};
+    node_buffer.set_nan();
+    node_buffer.set_value(PGM_def_input_node_id, node_id.data(), -1);
+    node_buffer.set_value(PGM_def_input_node_u_rated, node_u_rated.data(), -1);
+    input_dataset.add_buffer("node", 1, 1, nullptr, node_buffer);
+
+    std::vector<ID> const source_id{1};
+    std::vector<ID> const source_node{0};
+    std::vector<int8_t> const source_status{1};
+    std::vector<double> const source_u_ref{1.0};
+    std::vector<double> const source_sk{1000.0};
+    std::vector<double> const source_rx_ratio{0.0};
+    Buffer source_buffer{PGM_def_input_source, 1};
+    source_buffer.set_nan();
+    source_buffer.set_value(PGM_def_input_source_id, source_id.data(), -1);
+    source_buffer.set_value(PGM_def_input_source_node, source_node.data(), -1);
+    source_buffer.set_value(PGM_def_input_source_status, source_status.data(), -1);
+    source_buffer.set_value(PGM_def_input_source_u_ref, source_u_ref.data(), -1);
+    source_buffer.set_value(PGM_def_input_source_sk, source_sk.data(), -1);
+    source_buffer.set_value(PGM_def_input_source_rx_ratio, source_rx_ratio.data(), -1);
+    input_dataset.add_buffer("source", 1, 1, nullptr, source_buffer);
+
+    std::vector<ID> const sym_load_id{2};
+    std::vector<ID> const sym_load_node{0};
+    std::vector<int8_t> const sym_load_status{1};
+    std::vector<int8_t> const sym_load_type{2};
+    std::vector<double> const sym_load_p_specified{0.0};
+    std::vector<double> const sym_load_q_specified{500.0};
+    Buffer sym_load_buffer{PGM_def_input_sym_load, 1};
+    sym_load_buffer.set_nan();
+    sym_load_buffer.set_value(PGM_def_input_sym_load_id, sym_load_id.data(), -1);
+    sym_load_buffer.set_value(PGM_def_input_sym_load_node, sym_load_node.data(), -1);
+    sym_load_buffer.set_value(PGM_def_input_sym_load_status, sym_load_status.data(), -1);
+    sym_load_buffer.set_value(PGM_def_input_sym_load_type, sym_load_type.data(), -1);
+    sym_load_buffer.set_value(PGM_def_input_sym_load_p_specified, sym_load_p_specified.data(), -1);
+    sym_load_buffer.set_value(PGM_def_input_sym_load_q_specified, sym_load_q_specified.data(), -1);
+
+    if constexpr (std::is_same_v<input_type, row_t>) {
+        input_dataset.add_buffer("sym_load", 1, 1, nullptr, sym_load_buffer);
+    } else {
+        input_dataset.add_buffer("sym_load", 1, 1, nullptr, nullptr);
+        input_dataset.add_attribute_buffer("sym_load", "id", sym_load_id.data());
+        input_dataset.add_attribute_buffer("sym_load", "node", sym_load_node.data());
+        input_dataset.add_attribute_buffer("sym_load", "status", sym_load_status.data());
+        input_dataset.add_attribute_buffer("sym_load", "type", sym_load_type.data());
+        input_dataset.add_attribute_buffer("sym_load", "p_specified", sym_load_p_specified.data());
+        input_dataset.add_attribute_buffer("sym_load", "q_specified", sym_load_q_specified.data());
```

Comment:
instead, you can also add the `sym_load_buffer` initialization here. then, the only real difference between `row_t` and `columnar_t` is the call to `add_attribute_buffer` vs `set_value`

or will that cause `const` correctness clang-tidy warnings? I guess you can work around those by calling `set_nan` outside this code block

Reply:
I can move the initialization of Buffer outside while set_value remains inside the block